core/src/main/scala/org/apache/spark/metrics/ExecutorMetricType.scala

```diff
@@ -99,6 +100,61 @@ case object ProcessTreeMetrics extends ExecutorMetricType {
   }
 }
 
+case object GarbageCollectionMetrics extends ExecutorMetricType {
+  import GC_TYPE._
+  override val names = Seq(
+    "MinorGCCount",
+    "MinorGCTime",
+    "MajorGCCount",
+    "MajorGCTime"
+  )
+  override private[spark] def getMetricValues(memoryManager: MemoryManager): Array[Long] = {
+    var allMetrics = GCMetrics(0, 0, 0, 0)
+    ManagementFactory.getGarbageCollectorMXBeans.asScala.foreach { mxBean =>
+      val metrics = mxBean.getName match {
+        case `copy` | `psScavenge` | `parNew` | `g1Young` =>
```

Comment:
I am afraid that here is too version specific and implementation specfic. For example, how can we support OpenJDK 11 or OpenJ9?
Is there any way that it makes more configurable?

Reply:
this is a good point.  I don't think the GarbageCollectorMXBeans (or anything else in the JVM) define a mapping to major & minor GC.  So we could either:

a) manually update this mapping ourselves, as new GC methods come around.  also easy enough to make this configurable at runtime, so we don't have to cover all cases, but I think we'd still try to do as much as possible

b) don't map them into major / minor at all, and just report the metrics prefixed with the collector name.  This is a little tricky, as then the number of metrics is determined at runtime, based on what collectors are active in the given JVM.  But that should be manageable (it would wreck havoc if you had inconsistent JVMs across executors, though).

do you have thoughts on what might be more useful to the end user?

Reply:
I will try to use (a) to refactor it. I hope it could keep major/minor types.

Reply:
@kiszk I added some configurations to support more garbage collectors.